frontend/src/layouts/MainLayout.js

```diff
@@ -19,6 +19,7 @@ const MainLayout = observer(props => {
     drawer: {
       width: drawerWidth,
       flexShrink: 0,
+      zIndex: 1,
```

Comment:
do we need to set this? it locks the stacking context, and we should only do so if necessary.

Reply:
No, but we still need to apply z-index somewhere in order to get the spinner overlay to appear above the header and drawer. MUI has z-index values applied to the header and drawer by default, so I was overriding them. Instead I've moved the z-index property to the MainLayout div. This might be a better choice because it just enforces a stacking order between the main div and the snackbar/spinner which should always be over top of it anyway.